hydroflow_plus/src/location/mod.rs

```diff
@@ -30,18 +27,29 @@ pub use can_send::CanSend;
 pub mod tick;
 pub use tick::{NoTick, Tick};
 
-#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+#[derive(PartialEq, Eq, Clone, Debug)]
 pub enum LocationId {
     Process(usize),
     Cluster(usize),
+    Tick(usize, Box<LocationId>),
     ExternalProcess(usize),
 }
 
 impl LocationId {
+    pub fn root(&self) -> &LocationId {
+        match self {
+            LocationId::Process(_) => self,
+            LocationId::Cluster(_) => self,
+            LocationId::Tick(_, id) => id.root(),
+            LocationId::ExternalProcess(_) => self,
+        }
+    }
+
     pub fn raw_id(&self) -> usize {
         match self {
             LocationId::Process(id) => *id,
             LocationId::Cluster(id) => *id,
+            LocationId::Tick(_, _) => panic!("cannot get raw id for tick"),
             LocationId::ExternalProcess(id) => *id,
         }
     }
```

Comment:
fine for now but this enum seems a bit suspicious. `LocationId::Tick` is the location ID for a `Tick<Location>`? Weird that `LocationId` would be more than just an ID and instead contain a hierarchy inside it.

Reply:
Is `Tick<Tick<Location>>` a thing? (Is this a nested `loop`?)

Reply:
This is somewhat future-proofing; right now there is no way to create a `Tick<Tick<Location>>` but in the future that would be a nested loop.